# The Degenerate Remnants of Stars

## The Discovery of Sirius B
In 1838, [Friedrich Wilhelm Bessel](https://en.wikipedia.org/wiki/Friedrich_Bessel) used the technique of stellar parallax to find the distance to the star 61 Cygni.  The next candidate Bessel tried was [Sirus](https://en.wikipedia.org/wiki/Sirius), the brightest star in the sky.  It's parallax angle is $p = 0.379^{\prime\prime}$, which corresponds to a distance of only 2.64 pc, or 8.61 ly. Sirius' brilliance is due (in part) to its proximy to Earth.  Bessel tracked Sirius' path across the sky and found that it deviated slightly from a straight line.  Using 10 years of precise observation, Bessel concluded (in 1844) that Sirius was actually a binary star system.  He deduced that the binary orbital period was 50 years, where the modern value is $50.1284 \pm 0.0043$ years ([Bond et al 2017](https://iopscience.iop.org/article/10.3847/1538-4357/aa6af8)) and the position of the secondary star.

Bessel's telecopses were incapable of finding the "Pup" (i.e., the unseen companion of the luminous "Dog Star") due to the glare of Sirius.  After Bessel's death in 1846, the search for the Pup waned.  In 1862, [Alvan Graham Clark](https://en.wikipedia.org/wiki/Alvan_Graham_Clark) tested his father's new 18-inch refractor (3 inches longer than any previous instrument) on Sirius, and he discovered the Pup at its predicted position.  The dominant Sirius A was measured to be nearly $1000\times$ brighter than the Pup (now called Sirius B).  The orbits of the stars about their center of mass revealed that Sirius A and Sirius B are about $2.3\ M_\odot$ and $1.0\ M_\odot$, respectively.  The modern values are $2.063 \pm 0.023\ M_\odot$ and $1.018 \pm 0.011\ M_\odot$ ([Bond et al 2017](https://iopscience.iop.org/article/10.3847/1538-4357/aa6af8)).

Clark's discovery of Sirius B was made when the stars were most widely separated (by just $10^{\prime\prime}$) at apastron.  The great difference in their luminosities: $L_A = 25.4\ L_\odot$ and $L_B=0.056\ L_\odot$ makes observations difficult when the stars are closer together on the sky.  The next apastron comes 50 years later, where spectroscopists developed the tools to measure each star's surface temperature.  From the Pup's faint appearance, astronomers expected it to be cool and red, but they discovered that it is a hot, blue-white star that emits much of its energy in the UV.  Modern observations measure the surface temperature of Sirius B (the Pup) at $25,000 \pm 200\ {\rm K}$, which is much hotter than Sirius A at $9940\ {\rm K}$.

- **Given the modern measurements, what are the size and density of Sirius B?**

>The size of Sirius B can be estimated using the Stefan-Boltzman law,
>
>$$ \frac{L}{L_\odot} = \left(\frac{R}{R_\odot}\right)^2 \left(\frac{T}{T_\odot}\right)^4, $$
>
>where the Sun's surface temperature is $5777\ {\rm K}$.  Rearranging the above equation in terms of the stellar radius $R$, we get
>
>$$ \frac{R}{R_\odot} = \sqrt{\frac{L}{L_\odot}} \left(\frac{T}{T_\odot}\right)^{-2}. $$
>
>The size of Sirius B is 0.0126 $R_\odot$ or 1.38 $R_\oplus$, while the density is $7.11 \times 10^8\ {\rm kg/m^3}$.  Sirius B is a Solar mass star compressed into a roughly Earth-sized volume and the acceleration due to gravity is about $1.8 \times 10^6\ {\rm m/s^2}$.

In [7]:
import numpy as np
from scipy.constants import G

L_SiriusB = 0.056 #luminosity in solar units
T_SiriusB = 25000 #surface temperature of Sirius B in K
T_sun = 5777 #surface temperature of the Sun in K
R_sun = 6.957e8 #radius of the Sun in m
R_Earth = 6371e3 #radius of the Earth in m

R_SiriusB = np.sqrt(L_SiriusB)*(T_sun/T_SiriusB)**2

print("The radius of Sirius B is %1.4f R_Sun, %1.2f R_Earth, or %1.2e m." % (R_SiriusB,R_SiriusB*R_sun/R_Earth,R_SiriusB*R_sun))

M_sun = 1.989e30 #mass of the Sun in kg
M_SiriusB = 1.018 #mass of Sirius B in solar units

rho_SiriusB = (M_SiriusB*M_sun)/(4./3.*np.pi*(R_SiriusB*R_sun)**3)
print("The density of Sirius B is %1.2e kg/m^3" % rho_SiriusB)

g_SiriusB = G*M_SiriusB*M_sun/(R_SiriusB*R_sun)**2
print("The acceleration due to gravity at the surface of Sirius B is %1.2e m/s^2." % g_SiriusB)

The radius of Sirius B is 0.0126 R_Sun, 1.38 R_Earth, or 8.79e+06 m.
The density of Sirius B is 7.11e+08 kg/m^3
The acceleration due to gravity at the surface of Sirius B is 1.75e+06 m/s^2.


On Earth, the pull of gravity on a teaspoon of white dwarf material would be approximately $34,300\ {\rm N}$. This fierce gravity veals itself in the spectrum of Sirius B, where it produces an immense pressure near the surface that results in very broad hydrogen absorption lines.  Apart from these lines, its spectrum is a featureless continuum.

Astronomers initially dismissed the results, calling them "absurd".  However, the calculations are so simple and straightforward that this attitude changed.  In 1922, Eddington said, "*Strange objects, which persist in showing a type of spectrum entirely out of keeping with their luminosity, may ultimately teach us more than a host which radiate according to rule.*"  Like all sciences, astronomy advances most rapidly when confronted with exceptions to its theories.

## White Dwarfs
Obviously Sirius B is not a *normal* star, where it is in a class of stars that have approximately the mass of the Sun and the size of Earth called **white dwarfs**.  As many as 1/4 of the stars in the Solar neighborhood may be white dwarfs.  The average characteristics of these faint stars have been difficult to determine because a complete sample has only been obtained within 10 pc of the Sun.

### Classes of White Dwarf Stars

White dwarfs occupy a narrow sliver of the H-R diagram that is roughly parallel to (and below) the main sequence.  Although white dwarfs are typically whiter than normal stars, the name itself is a misnomer since they come in all colors, with surface temperatures ranging from $5000-80,000\ {\rm K}$.  Their spectral type D has several subdivisions, 

-  the largest group (about 2/3 of the total number, including Sirius B) is called **DA white dwarfs** and display only pressure-broadened hydrogen absorption lines in their spectra. 
- Hydrogen lines are absent from the **DB white dwarfs** (8% of the total number), which show only helium absorption lines.
- No lines are shown at all in the **DC white dwarfs** (14% of the total number), only a continuum devoid of features.
- The remaining 11% include the **DQ white dwarfs** that exhibit carbon features in their spectra and **DZ white white dwarfs** that have evidence of metal lines.

### Central Conditions in White Dwarfs
White dwarfs are very massive for their size, where we can not estimate the central pressure and temperature using our derived values.  Assuming that Sirius B represents a typical white dwarf (at least 2/3 of them) and recalling the estimate for the central pressure from [hydrostatic equilibrium](https://saturnaxis.github.io/ModernAstro/Chapter_11/interiors-of-stars.html#the-derivation-of-the-hydrostatic-equilibrium-equation), we have

$$ P_c = \int_{P_c}^0 dP = \int_0^{R_{wd}} G\left(\frac{4}{3}\pi r^3 \rho \right)\rho rdr = \frac{2}{3}\pi G \rho_{wd}^2 R_{wd}^2,  \tag{1}$$

which is roughly $5.5 \times 10^{21}\ {\rm N/m^2}$ or 200,000 times greater than the pressure in the Sun $(P_{c,\odot} = 2.477 \times 10^{16}\ {\rm N/m^2})$.

In [22]:
import numpy as np
from scipy.constants import G

M_sun = 1.989e30 #mass of the Sun in kg
T_sun = 5777 #surface temperature of the Sun in K
R_sun = 6.957e8 #radius of the Sun in m

M_SiriusB = 1.018*M_sun #mass of Sirius B in kg
L_SiriusB = 0.056 #luminosity in solar units
T_SiriusB = 25000 #surface temperature of Sirius B in K

R_SiriusB = np.sqrt(L_SiriusB)*(T_sun/T_SiriusB)**2*R_sun #radius of Sirius B in m

rho_SiriusB = M_SiriusB/(4./3.*np.pi*R_SiriusB**3)

P_c = (2./3.)*G*np.pi*rho_SiriusB**2*R_SiriusB**2
print("The central pressure in Sirius B is %1.2e N/m^2." % P_c)

P_sun = 2.477e+16 #central pressure in Sun in N/m^2
print("The central pressure in Sirius B is %1.0e times greater than the Sun." % (P_c/P_sun))

The central density of Sirius B is 7.11e+08 kg/m^3.
The central pressure in Sirius B is 5.47e+21 N/m^2.
The central pressure in Sirius B is 2e+05 times greater than the Sun.


A crude estimate of the central temperature can be obtained for the [radiative temperature gradient](https://saturnaxis.github.io/ModernAstro/Chapter_11/interiors-of-stars.html#the-radiative-temperature-gradient) as

$$ \frac{dT}{dr} = \frac{T_{wd}-T_c}{R_{wd}-0} = - \frac{3}{4ac}\frac{\overline{\kappa}\rho}{T_c^3}\frac{L_{wd}}{4\pi R_{wd}^2},  $$

assuming that the surface temperature $T_{wd} \ll T_c$.  We can also assume that $\overline{\kappa} = 0.02\ {\rm m^2/kg}$ (from the [Rosseland mean](https://saturnaxis.github.io/ModernAstro/Chapter_9/stellar-atmospheres.html#the-rosseland-mean-opacity) opacity for electron scattering with $X=0$), which gives

$$ T_c \approx \left[\frac{3\overline{\kappa}\rho}{4ac} \frac{L_{wd}}{4\pi R_{wd}} \right]^{1/4} \approx 5.5 \times 10^7\ {\rm K}.$$


In [18]:
import numpy as np
from scipy.constants import G,sigma,c

M_sun = 1.989e30 #mass of the Sun in kg
T_sun = 5777 #surface temperature of the Sun in K
R_sun = 6.957e8 #radius of the Sun in m
L_sun = 3.827e26 #luminosity of the Sun in W

M_SiriusB = 1.018*M_sun #mass of Sirius B in kg
L_SiriusB = 0.056*L_sun #luminosity in solar units
T_SiriusB = 25000 #surface temperature of Sirius B in K

R_SiriusB = np.sqrt(L_SiriusB/L_sun)*(T_sun/T_SiriusB)**2*R_sun #radius of Sirius B in m
rho_SiriusB = M_SiriusB/(4./3.*np.pi*R_SiriusB**3)
kappa_es = 0.02
a_rad = 4.*sigma/c

T_c = ((3*kappa_es*rho_SiriusB/(4*a_rad*c))*(L_SiriusB/(4.*np.pi*R_SiriusB)))**0.25

print("The central temperature in Sirius B is %1.2e K." % T_c)

The central temperature in Sirius B is 5.50e+07 K.


These estimated values for a white dwarf show that hydrogen must be absent from the subsurface layers because they are high enough to start fusion and the white dwarf luminosities would be several orders of magnitude larger than observed.  As a result, the interiors of white dwarfs must contain particles that are incapable of fusion at these densities and temperatures (e.g., constituents of the PP I chain).

White dwarfs are born from the cores of low- and intermediate-mass stars (i.e., those with a ZAMS mass $\lesssim 8-9\ M_\odot$) near the end of their lives on the AGB of the H-R diagram.  Any star with a helium core mass $\gtrsim 0.5\ M_\odot$ will undergo fusion and thus, most white dwarfs consist primarily of completely ionized carbon and oxygen nuclei (i.e., the left over ${\rm CO}$ core).  As an AGB star ages, it expels its surface layers as planetary nebula, which exposes the core as a white dwarf.  The distribution of DA white dwarf masses is sharply peaked at $0.56\ M_\odot$, with some 80% lying between $0.42-0.70\ M_\odot$.  The much larger main-sequence masses must be lost on the AGB through thermal pulses and a superwind.

### Spectra and Surface Composition
The exceptionally strong pull of a white dwarf's gravity is responsible for the characteristic spectrum of DA white dwarfs.  In this process, heavier nuclei are pulled below the surface while the lighter hydrogen rises to the top and results in a thin outer layer of hydrogen covering a helium layer on top of the ${\rm CO}$ core.  This vertical stratification takes only $\sim100$ years in the hot stellar atmosphere.

The origin of the non-DA (e.g., DB and DC) white dwarfs is not yet clear.  Efficient mass-loss may occur on the AGB associated with the thermal pulse or superwind phases, which strip the white dwarf of nearly all of its hydrogen.  Alternatively, a single white dwarf may be transformed between the DA and non-DA spectral types through convective mixing.  For example, the helium convection zone's penetration into a thin hydrogen layer could change a DA into a DB white dwarf by diluting the hydrogen with additional helium. 

### Pulsating White Dwarfs

White dwarfs with surface temperatures of $T_e \approx 12,000\ {\rm K}$ lie within the instability strip of the H-R diagram and pulsate with periods between $100-1000\ {\rm s}$.  In 1968, Arlo Landolt discovered a class of variable white dwarf stars, which are known as **ZZ Ceti** variables or **DAV** stars.  The pulsation periods correspond to the nonradial *g*-modes that resonate within the white dwarf's surface layers of hydrogen and helium.  These *g*-modes involve almost perfectly horizontal displacement and thus, the radii of these compact pulsators hardly change.  Their brightness variations ($\sim 0.1\ {\rm mag}$) are due to temperature variations on the stellar surface.

Successful numerical calculations of pulsating white dwarf models were carried out by [Don Winget (and others)](https://ui.adsabs.harvard.edu/abs/2008ARA%26A..46..157W/abstract), where they demonstrated that the *hydrogen* partial ionization zone is responsible for driving oscillations in the ZZ Ceti stars.  These computations also confirmed the elemental stratification of white dwarf envelopes.  These numerical models predict that hotter DB white dwarfs should also exhibit *g*-mode oscillations driven by the *helium* partial ionization zone.  Within a year's time from the initial prediction, it was confirmed when the first **DBV** star was discovered.  Some stars are described by their transition to the white dwarf stage as: DOV and PNNVs stars that have an effective temperature $T_e \approx 10^5\ {\rm K}$.  The DO spectral type marks the initial stage of the white dwarf and the PNNV stands for planetary nebula nuclei.

## The Physics of Degenerate Matter

### The Pauli Exclusion Principle and Electron Degeneracy
Any system consists of quantum states that are identified by a set of quantum numbers.  A box of gas particles is filled with standing de Broglie waves that are described by three quantum numbers specifying a particle's momentum in each of three directions.  If the gas particles are fermions (e.g., electrons or neutrons), then the Pauli exclusion principle allows at most one fermion in each quantum state because no two fermions can have the same set of quantum numbers.

At standard temperature and pressure ($20\ {\rm ^\circ C}$ and 101,325 ${\rm N/m^2}$), only one of every $10^7$ quantum states is occupied by a gas particle, and the limitations imposed by the Pauli exclusion principle become insignificant.  Ordinary gas has a *thermal* pressure that is related to it temperature by the ideal gas law (i.e., $PV = NkT$).  As energy ($PV$) is removed from the gas and its temperature falls, an increasingly large fraction of the particles are forced into *lower* energy states.  If the particles are fermions, only one particle is allowed in each state and thus, all the particles cannot crowd into the ground state.

As the temperature of the gas is lowered, the fermions will fill up the lowest available unoccupied state first and successively occupy the excited states with the lowest energy.  Even in as the temperature approaches absolute zero ($T\rightarrow 0\ {\rm K}$), the motion of the fermions in excited states produces a gas pressure.  At absolute zero, *all** the lower energy states and *none* of the higher energy states are occupied.  In this case, the fermion gas is completely **degenerate**.

### The Fermi Energy

The maximum energy of any electron in a completely degenerate gas (at absolute zero) is known as the **Fermi energy** $\varepsilon_F$.  To determine this limiting energy, **consider a 3D box of length L on each side.** Treating the electrons as standing waves in the box, their wavelengths in each dimension are,

$$ \lambda_x = \frac{2L}{N_x}, \qquad \lambda_y = \frac{2L}{N_y}, \qquad \lambda_z = \frac{2L}{N_z}, $$

where there are integer quantum numbers ($N_x$, $N_y$, and $N_z$) associated with each dimension.  Recalling that the de Broglie wavelength is related to the components of momentum by,

$$ p_x = \frac{hN_x}{2L}, \qquad p_y = \frac{hN_y}{2L}, \qquad p_z = \frac{hN_z}{2L}. $$

The total kinetic energy of a particle is the sum of the squares of each component of the momentum, or

$$ \varepsilon = \frac{p^2}{2m} = \frac{1}{2m}\left(p_x^2 + p_y^2 + p_z^2 \right). $$

In terms of all the fundamental constants, the energy is

$$ \varepsilon = \frac{h^2}{8mL^2}\left(N_x^2 + N_y^2 + N_z^2 \right) = \frac{h^2N^2}{8mL^2}. \tag{2}$$

The total number of electrons in the gas corresponds to the total number of unique quantum numbers $(N_x,\ N_y,\ \text{and}\ N_z)$ times two.  

```{note}
The factor two arises due to the electrons being spin $1/2$ particles, so $m_s \pm 1/2$ implies that two electron can have the same combination of $N_x,\ N_y,\ \text{and}\ N_z$, while still maintaining a unique set of *four* quantum numbers.
```

Each integer in a coordinate *N*-space corresponds to the quantum state of two electrons.  With a large enough sample of electrons, the electrons will fill out up to a radius $N$ but only for the positive octant of *N*-space (i.e., $N_x>0$, $N_y>0$, and $N_z>0$).  this means that the total number of electrons will be 

$$ N_e = 2 \times \left(\text{volume of a positive octant of a sphere of radius N}\right) = 2\left(\frac{1}{8}\right)\left( \frac{4}{3}\pi N^3\right). $$

Solving for $N$ and substituting into Eq. 2, we find that the Fermi energy is given by

$$ \varepsilon_F = \frac{\hbar^2}{2m_e} \left( 3\pi^2 n\right)^{2/3}, \tag{3} $$

which depends on the mass of the electron $m_e$ and the number of electrons per unit volume $(n \equiv N_e/L^3)$.  The average energy per electron at absolute zero is $\frac{3}{5}\varepsilon_F$.  The above derivation applies to any fermion, not just electrons.

### The Condition for Degeneracy
Some of the states with an energy less than $\varepsilon_F$ will become vacant as fermions use their thermal energy to occupy more energetic states.  Although the degeneracy will not be precisely complete when $T>0\ {\rm K}$, the assumption of complete degeneracy is a good approximation at the densities encountered in the interior of a white dwarf.  All but the most energetic particles will have an energy less than the Fermi energy

To understand the dependency of degeneracy on both temperature and density, we express the Fermi energy in terms of the density of the electron gas.  For full ionization, the number of electrons per unit volume is

$$ n_e = \left(\frac{\text{# of electrons}}{\text{nucleon}}\right) \left(\frac{\text{# of nucleons}}{\text{volume}} \right) = \left( \frac{Z}{A}\right) \frac{\rho}{m_H}, \tag{4} $$

which depends on the number of protons $Z$, nucleons $A$, and the mass of the hydrogen atom $m_H$.  The Fermi energy is proportional to the $2/3$ power of the density,

$$ \varepsilon_F = \frac{\hbar^2}{2m_e} \left[ 3\pi^2 \left( \frac{Z}{A}\right) \frac{\rho}{m_H}\right]^{2/3}. \tag{5} $$

Compare the Fermi energy with the average thermal energy of an electron ($\frac{3}{2}kT$).  In rough terms, if $\frac{3}{2}kT < \varepsilon_F$, then an average electron wil be unable to make a transition to an unoccupied state, and the electron will be degenerate.  For a degenerate gas,

$$ \frac{3}{2}kT < \frac{\hbar^2}{2m_e} \left[ 3\pi^2 \left( \frac{Z}{A}\right) \frac{\rho}{m_H}\right]^{2/3}, $$

or

$$\frac{T}{\rho^{2/3}} < \frac{\hbar^2}{3m_e k}\left[ \frac{3\pi^2}{m_H}\left(\frac{Z}{A}\right) \right]^{2/3}. $$

For $Z/A = 0.5$, the right-hand side is $\mathcal{D} \equiv 1261\ {\rm K\ m^2/kg^{2/3}}$, so that the condition for degeneracy can be written as

$$ \frac{T}{\rho^{2/3}} < \mathcal{D}. \tag{6}$$

The smaller the value of $T/\rho^{2/3}$, the more degenerate the gas.

- **How important is electron degeneracy at the centers of the Sun and Sirius B?**

> At the center of the standard solar model, $T_c = 1.57 \times 10^7\ {\rm K}$ and $\rho_c = 1.527 \times 10^5\ {\rm kg/m^3}$.  Then
>
> $$ \frac{T_c}{\rho^{2/3}} = 5500\ {\rm K\ m^2/kg^{2/3}} > \mathcal{D}. $$
>
>In the Sun, electron degeneracy is quite weak and plays a very minor role supplying only $\sim 0.1\%$ of the central pressure.  However, as the Sun continues to evolve, electron degeneracy will become increasingly important.
>
>For Sirius B, the values of the density and central temperature estimated above lead to 
>
> $$ \frac{T_c}{\rho^{2/3}} = 69\ {\rm K\ m^2/kg^{2/3}} \ll \mathcal{D}, $$
>
>which validates the assumption of complete degeneracy for Sirius B

In [49]:
import numpy as np
from scipy.constants import G,sigma,c

rho_sun_c = 1.527e5 #central density of the Sun in kg/m^3
T_sun_c = 1.57e7 #central temperature fo the Sun in K
degen_sun = T_sun_c/rho_sun_c**(2/3)

rho_SiriusB_c = 7.11e8 #central density of Sirius B in kg/m^3
T_SiriusB_c = 5.50e7 # central temperature for Sirius B in K 
degen_SiriusB = T_SiriusB_c/rho_SiriusB_c**(2/3)

print("The condition for degeneracy is %d K m^2/kg^{2/3} for the Sun." % degen_sun)
print("The condition for degeneracy is %d K m^2/kg^{2/3} for Sirius B." % degen_SiriusB)

The condition for degeneracy is 5495 K m^2/kg^{2/3} for the Sun.
The condition for degeneracy is 69 K m^2/kg^{2/3} for Sirius B.


### Electron Degeneracy Pressure
We can estimate the electron degeneracy pressure by combining two key ideas of quantum mechanics:

1. The Pauli exclusion principle, which allows at most one electron in each quantum state, and
2. Heisenberg's uncertainty principle in the form of, $\Delta x\ \Delta p_x \approx \hbar$, which requires that an electron confined to small volume of space have a correspondingly high uncertainty in its momentum.

When we make the *unrealistic assumption* that all of the electrons have the same momentum $p$, the pressure integral becomes

$$ P \approx \frac{1}{3}n_e\ p\ v, \tag{7}$$

which depends on the total electron number density $n_e$ and average speed $v$ of the electrons.

In a completely degenerate electron gas and for a uniform number density of $n_e$, the separation between neighboring electrons is about $n_e^{-1/3}$.  However, the electron must maintain their identities as different particles to satisfy the Pauli exclusion principle (i.e., the uncertainty in their positions cannot be larger than their physical separation).  Identifying $\Delta x \approx n_e^{-1/3}$ for the limiting case of complete degeneracy, we can use Heisenberg's uncertainty relation to estimate the momentum of an electron.  In on coordinate direction,

$$ p_x \approx \Delta p_x \approx \frac{\hbar}{\Delta x} \approx \hbar n_e^{1/3}. \tag{8} $$

In a 3D gas each of the directions is equally likely, which implies that

$$ p_x^2 = p_y^2 = p_z^2, $$

and is just a statement of the equipartition of energy among all the coordinate directions.  Therefore,

$$ p^2 = p_x^2 = p_y^2 = p_z^2 = 3 p_x^2, $$

or

$$ p = \sqrt{3}p_x. $$

Using Eqn. 4 for the electron number density with full ionization gives

$$ p \approx \sqrt{3}\hbar \left[\left(\frac{Z}{A}\right)\frac{\rho}{m_H}\right]^{1/3}. $$

For *nonrelativistic* electrons, the speed is 

$$ v = \frac{p}{m_e} \approx \frac{\sqrt{3}\hbar}{m_e}\left[\left(\frac{Z}{A}\right)\frac{\rho}{m_H}\right]^{1/3}. \tag{9} $$

Inserting Eqns. 4 and 8 into Eqn. 9 for for the electron degeneracy pressure results in

$$ P \approx \frac{\hbar^2}{m_e}\left[\left(\frac{Z}{A}\right)\frac{\rho}{m_H}\right]^{5/3}. \tag{10} $$

This is roughly a factor of tow smaller than the exact expression for the pressure due to a completely degenerate, nonrelativistic electron gas $P$,

$$ P = \frac{\left(3\pi^2\right)^{2/3}}{5} \frac{\hbar^2}{m_e}\left[\left(\frac{Z}{A}\right)\frac{\rho}{m_H}\right]^{5/3}. \tag{11} $$

Using $Z/A = 0.5$ for a carbon-oxygen white dwarf, Eqn. 11 shows that the electron degeneracy pressure available to support Sirius B is about $1.9 \times 10^{22}\ {\rm N/m^2}$, which is within a factor of two from the central pressure estimated in Eqn. 1. *Electron degeneracy pressure is responsible for maintaining the hydrostatic equilibrium in a white dwarf*.

```{note}
You may have noticed that Eqn. 11 is the polytropic equation of state, $P=K\rho^{5/3}$ with the polytropic index $n = 3/2$.  This implies that the Lane-Emden equation can be used to study these objects. 
```

## The Chandrasekhar Limit

### The Mass-Volume Relation
The relation between the radius $R_{wd}$ and mass $M_{wd}$ of a white dwarf can be found by setting the estimate of the central pressure (Eqn. 1) equal to the electron degeneracy pressure (Eqn. 11):

$$ \frac{2}{3} \pi G\rho^2 R_{wd}^2 = \frac{\left(3\pi^2\right)^{2/3}}{5} \frac{\hbar^2}{m_e}\left[\left(\frac{Z}{A}\right)\frac{\rho}{m_H}\right]^{5/3}. $$

Assuming a constant density for the white dwarfs leads to an estimate of the radius,

$$ R_{wd} \approx \frac{\left(18\pi \right)^{2/3}}{10} \frac{\hbar^2}{GM_{wd}^{1/3}m_e} \left[\left(\frac{Z}{A}\right)\frac{\rho}{m_H}\right]^{5/3}. \tag{12} $$

For a Solar-mass carbon-oxygen white dwarf, $R_{wd} \approx 2.9 \times 10^6\ {\rm m}$, which is too small by roughly a factor of 2 but an acceptable estimate.  A surprising implication is that $M_{wd}R_{wd}^3 = \text{constant}$, or 

$$ M_{wd}V_{wd} = \text{constant}. \tag{13} $$

The volume of a white dwarf is inversely proportional to its mass, where *more massive white dwarfs are actually smaller*.  This **mass-volume relation** is a result of the electron degeneracy pressure supporting the star.  The electrons must be more closely confined to generate the larger degeneracy pressure required to support a more massive star (i.e., $\rho \propto M_{wd}^2$).

According to the mass-volume relation, a white dwarf would eventually shrink down to zero volume in the limit of continually adding mass.  However, if the density exceeds about $10^9\ {\rm kg/m^3}$, then there is a departure from this relation.  Using Eqn. 9 to estimate the speed of electrons in Sirius B results in

$$ v = \frac{\hbar}{m_e}\left[\left(\frac{Z}{A}\right)\frac{\rho}{m_H}\right]^{1/3} = 6.91 \times 10^7\ {\rm m/s}, $$

which is $0.23c$.  

In [66]:
from scipy.constants import hbar, m_e,c

Z = 1.
A = 2.
m_H = 1.6735575e-27 #mass of hydrogen in kg
rho_SiriusB = 7.11e8 #central density of Sirius B in kg/m^3

v_e =(hbar/m_e)*((Z/A)*(rho_SiriusB/m_H))**(1./3.)
print("The average speed of electrons in Sirius B is %1.2e m/s or %1.3f c." % (v_e,v_e/c))

The average speed of electrons in Sirius B is 6.91e+07 m/s or 0.230 c.


If the mass-volume relation were correct, white dwarfs a bit more massive than Sirius B would be so mall and dense that their electrons would be superluminal (i.e., faster than the speed of light).  This points to the dangers of ignoring relativity in our expressions for the electron speed and pressure.  Because the electrons are moving more slowly than the nonrelativistic Eqn. 10 would indicate, there is less electron pressure available to support the star.  Thus, a massive white dwarf is smaller than predicted by the mass-volume relation.  *There is a limit to the amount of matter than can be supported by electron degeneracy pressure*.

### Dynamical Instability
To appreciate the effect of relativity on the stability of a white dwarf, recall that Eqn. 11 is a polytropic equation ($P = K\rho^{5/3}$).  The ratio of the specific heats $\gamma = 5/3$ in the nonrelativistic limit.  This means that the white dwarf is dynamically stable.  If it suffers a small perturbation, then it will return to its equilibrium structure instead of collapsing.  In the extreme relativistic limit, the electron speed $v=c$ must be used to find the degeneracy pressure, which results in

$$ P = \frac{\left(3\pi^2\right)^{1/3}}{4} \hbar c\left[\left(\frac{Z}{A}\right)\frac{\rho}{m_H}\right]^{4/3}. \tag{14}$$

In this limit, $\gamma = 4/3$ corresponds to dynamical instability and introduces a polytropic equation with an index $n=3$.  The smallest departure from equilibrium will cause the white dwarf to collapse as the electron degeneracy pressure fails.  Approaching this limiting case leads to the collapse of the degenerate core in an aging supergiant and results in a core-collapse supernova.

### Estimating the Chandrasekhar Limit
An approximate value for the maximum white dwarf mass can be obtained by setting Eqn. 14 equal to the estimate of the central pressure (Eqn. 1) to get

$$ \frac{2}{3} \pi G \rho_{wd}^2 R_{wd}^2  = \frac{\left(3\pi^2\right)^{1/3}}{4} \hbar c\left[\left(\frac{Z}{A}\right)\frac{\rho_{wd}}{m_H}\right]^{4/3}, $$

and solving for $\rho$ to get,

$$ \rho_{wd}^{2/3} = \frac{\hbar c}{8\pi^{1/3} G R_{wd}^2  } \left[\left(\frac{Z}{A}\right)\frac{3}{m_H}\right]^{4/3}. $$

Using $\rho = M_{wd}/\frac{4}{3}\pi R_{wd}^3$ gives

$$ M_{wd}^{2/3} = \left(\frac{4\pi}{3}\right)^{2/3} \frac{\hbar c}{8\pi^{1/3} G } \left[\left(\frac{Z}{A}\right)\frac{3}{m_H}\right]^{4/3}, $$

and

$$ M_{Ch} = \left(\frac{3\sqrt{2\pi}}{8}\right) \left(\frac{\hbar c}{ G } \right)^{3/2} \left[\left(\frac{Z}{A}\right)\frac{1}{m_H}\right]^2. \tag{15} $$

Using $Z/A = 0.5$ results in $0.44\ M_\odot$ for the greatest possible mass.  Note that Eqn. 15 contains the fundamental constants $\hbar$, $c$, and $G$, which represent  the combined effects of quantum mechanics relativity, and Newtonian gravitation on the structure of a white dwarf. A [precise derivation with the Lane-Emden equation](https://scholar.harvard.edu/files/schwartz/files/15-stars.pdf) results in a value of $M_{ch} \approx 1.4\ M_\odot$, which is the **Chandrasekhar limit**.

It is important to emphasize that the gas pressure of the ideal gas law and the expression for radiation pressure depend explicitly on temperature, where the pressure of a completely degenerate electron gas is independent of its temperature.  This has the effect of decoupling the mechanical structure of the star from its thermal properties.

## The Cooling of White Dwarfs
Most stars end their lives as white dwarfs, where no fusion occurs in their interiors.  White dwarfs simply cool off at an essentially constant radius as they slowly deplete their supply of thermal energy.  Just as paleontologists can read the history of Earth's life in the fossil record, astronomers try to recover the history of star formation by studying the statistics of white-dwarf temperatures.

### Energy Transport
"How is energy transported outward from the interior of a white dwarf?" In an ordinary star, photons travel much farther than atoms do before a collision that robs them of energy.  As a result, photons are normally more efficient carriers of energy to the stellar surface.  In a white dwarf, the degenerate electrons can travel long distances before losing energy in a collision with a nucleus because the vast majority of the lower-energy electron states are already occupied.  Thus, energy is carried by **electron conduction** rather than by radiation.

This process is so efficient that hte interior of a white dwarf is nearly isothermal, with the temperature dropping significantly only in the nondegenerate surface layers.  The steep temperature gradient near the surface creates convection zones that may alter the appearance of the white dwarf's spectrum as it cools.

At the surface, a white dwarf has a luminosity $L_{wd}$ and mass $M_{wd}$ that contribute to a surface pressure $P$ as a function of the temperature because the envelope is nondegenerate.  The equation for this surface pressure is

$$ P = \left( \frac{4}{17} \frac{16\pi ac}{3} \frac{GM_{wd}}{L_{wd}} \frac{k}{\kappa_o \mu m_H} \right)^{1/2} T^{17/4}, \tag{16} $$

which depends on the coefficient of the bound-free Kramers opacity law

$$ \kappa_o = 4.34 \times 10^{21}\ Z(1+X)\ {\rm m^2/kg}. $$

Using the ideal gas law to replace the pressure results in a relation between the density and temperature,

$$ \rho = \left( \frac{4}{17} \frac{16\pi ac}{3} \frac{GM_{wd}}{L_{wd}} \frac{\mu m_H}{\kappa_o k} \right)^{1/2} T^{13/4}. \tag{17}$$

The transition between the nondegenerate surface layers of the star and its *isothermal degenerate interior* of temperature $T_c$ is described by using Eqn. 17 in Eqn. 6 by finding

$$ \rho^{2/3} = \left( \frac{4}{17} \frac{16\pi ac}{3} \frac{GM_{wd}}{L_{wd}} \frac{\mu m_H}{\kappa_o k} \right)^{1/3} T^{13/6} $$

and substituting into $\mathcal{D} = T/\rho^{2/3}$ and solving $L_{wd}$ by

$$ L_{wd} = \frac{4}{17} \frac{16\pi ac}{3} \frac{GM_{wd}}{L_{wd}} \frac{\mu m_H}{\kappa_o k} T_c^{7/2} = C T_c^{7/2}, \tag{18} $$

where 

$$ C \equiv 6.65 \times 10^{-3} \left(\frac{M_{wd}}{M_\odot} \right)\frac{\mu}{Z(1+X)}. $$

Note that the luminosity is proportional to the *interior* temperature $T_c^{7/2}$ and that is varies to the fourth power of the *effective* temperature according to the Stefan-Boltzmann law.  Thus, the surface of a white dwarf cools more slowly than its isothermal interior as the the star's thermal energy leaks into space (i.e., $\frac{7}{2} < 4$).

- **What is the interior temperature of a Solar-mass white dwarf with $L_{wd} =0.03 L_\odot$?**  Assume that $X = 0,\ Y=0.9,\ \text{and}\ Z=0.1$ for the nondegenerate envelop so $\mu \simeq 1.4$.

>Using Eqn. 18 and solving for $T_c$, we get
>
>$$ T_c = \left[ \frac{L_{wd}}{6.65 \times 10^{-3}} \left(\frac{M_{wd}}{M_\odot}\right)^{-1} \frac{Z(1+X)}{\mu} \right]^{2/7} = 2.85 \times 10^8\ {\rm K}.$$
>
>Substituting into the equation for the degeneracy condition shows that the density at the base of the nondegenerate envelope is about
>
>$$ \rho = \left(\frac{T_c}{\mathcal{D}} \right)^{3/2} = 3.4 \times 10^6\ {\rm kg/m^3}. $$
>
>This result is several orders of magnitude less than the average density of a Solar-mass white dwarf (e.g., Sirius B).  It also confirms that the envelope is indeed thin and contributes very little to the star's total mass.


In [71]:
L_sun = 3.827e26 #luminosity of the Sun in W
L_wd = 0.03*L_sun #Luminosity of the white dwarf in W
M_wd = 1. #Solar-mass white dwarf

Z = 0.1
X = 0
mu = 1.4

T_c = ((L_wd/6.65e-3)*(1./M_wd)*(Z*(1+X)/mu))**(2/7)
print("The interior temperature of the white dwarf is %1.2e K." % T_c)

D = 1261 # degeneracy condition in K m^2/kg^(2/3)
rho = (T_c/D)**1.5

print("The density at the base of the nondegenerate envelope is %1.2e kg/m^3." % rho)

The interior temperature of the white dwarf is 2.85e+07 K.
The density at the base of the nondegenerate envelope is 3.39e+06 kg/m^3.


### The Cooling Timescale
A white dwarf's thermal energy primarily resides in this kinetic energy of its nuclei, where the degenerate electrons cannot give up a significant amount of energy nearly all the lower energy state are already occupied.  If we assume for simplicity that the composition is uniform, then the total number of nuclei in the white dwarf is equal to the star's mass $M_{wd}$ divided by the mass of a nucleus $Am_H$.  Since the average thermal energy of a nucleus is $\frac{3}{2}kT$, the thermal energy available for radiation is

$$ U = \frac{M_{wd}}{Am_H}\frac{3}{2}kT_c \tag{19}. $$

If use $T_c = 2.8\times 10^7\ {\rm K}$ and $A = 12$ for carbon, then Eqn. 19 gives approximately $6 \times 10^{40}\ {\rm J}$.  A crude estimate of the characteristic timescale for cooling $\tau_{cool}$ can be made simply by dividing the thermal energy by the luminosity.  Thus

$$ \tau_{cool} = \frac{U}{L_{wd}} = \frac{3M_{wd}k}{2Am_H C T_c^{5/2}}, \tag{20}$$

which is about $5.2 \times 10^{15}\ {\rm s} \approx 170\ {\rm Myr}$.  This is an underestimate, because the cooling timescale increases as $T_c$ decreases.

### The Change in Luminosity with Time
The depletion of the internal energy provides the luminosity, so Eqns. 18 and 19 give

$$ \frac{-dU}{dt} = L_{wd} $$

or 

$4 -\frac{d}{dt}\left( \frac{M_{wd}}{Am_H}\frac{3}{2}kT_c \right) = CT^{7/2}.$$

If the initial temperature of the interior is $T_o$, then this expression may be integrated to obtain the core temperature as a function of time:

$$ T_c(t) = T_o \left(1 + \frac{5}{3}\frac{Am_H C T_o^{5/2}}{M_{wd}k}t \right)^{-2/5} = T_o\left(1 + \frac{5t}{2\tau_o} \right)^{-2/5}, \tag{21}$$

where the timescale for cooling at the initial temperature is $\tau_o \equiv \frac{3M_{wd}k}{2Am_H C T_o^{5/2}}$.  Using the same logic for Eqn. 18 shows that the luminosity of the white dwarf declines sharply from its initial value $(L_o = C T_o^{7/2})$ and then dims much more gradually as time passes:

$$ L_{wd} = L_o \left(1 + \frac{5}{3}\frac{Am_H C L_o^{5/2}}{M_{wd}k}t \right)^{-2/5} = L_o\left(1 + \frac{5t}{2\tau_o} \right)^{-2/5}. \tag{22} $$

### Crystallization
As a white dwarf cools, it *crystallizes* in a gradual inside-out process.  An upturned "knee" occurs in theoretical cooling curves at about $L_{wd}/L_\odot \approx 10^{-4}$ when the cooling nuclei begin settling into a crystalline lattice.  The regular crystal structure is maintained by the mutual electrostatic repulsion of the nuclei, where it minimizes their energy as they vibrate about their average position in the lattice.

As the nuclei undergo this phase change, they release their latent heat (about $kT$ per nucleus), which slows the star's cooling and produces the knee in the cooling curve.  Later as the white dwarf's temperature continues to drop, the crystalline lattice actually accelerates the cooling as the coherent vibration of the regularly spaced nuclei promotes further energy loss.  This is reflected in the subsequent downturn in the cooling curve.  The ultimate monument to the lives of most stars will be a *diamond in the sky*.  Less poetically, it is a cold, dark, Earth-sized sphere of crystallized carbon and oxygen floating through the depths of space.  Note that the body-centered cubic lattice of the white dwarf's nuclei is more similar to metallic sodium than terrestrial diamond.

### Comparing Theory with Observations
It is possible to observe the cooling of a pulsating white dwarf despite the large uncertainties in the measurement of its surface temperature that result from high surface gravity and broad spectral features.  As the star's temperature declines, its period $P$ slowly changes (approximately) according to $dP/dt \propto T^{-1}$.  Extremely precise measurements of a rapidly cooling ${\rm DOV}$ star yield $P/|dP/dt| = 1.4 \times 10^6\ {\rm years}$, in excellent agreement with the theoretical value.  Measuring period changes for the more slowly cooling ${\rm DBV}$ and ${\rm DAV}$ stars are even more difficult.

This interest in an accurate calculation of the decline in a white dwarf's temperature reflects the motivation for using these "fossil" stars as a tool for uncovering the history of star formation in our Galaxy.  [Winget et al (1987)](https://ui.adsabs.harvard.edu/abs/1987ApJ...315L..77W/abstract) illustrate how this might be accomplished by plotting the number of observed white dwarfs per cubic parsec as a function of their brightness (either luminosity or absolute magnitude).

There is a dramatically sudden drop in the number of white dwarfs with $L_{wd}/L_\odot < -4.5$ (or $M_V \gtrsim 16$) that is inconsistent with the assumption that stars have been forming in our Galaxy throughout the infinite past.  Instead, this decline is best explained if the first white dwarfs were formed and began cooling $9.0 \pm 1.8\ {\rm Gyr}$ ago.  Combining the theoretical models with observations and adding the time spent in the pre-white dwarf stages of stellar evolution implies that star formation in the disk of our Galaxy began about $9.3 \pm 2.0\ {\rm Gyr}$ ago.  This time is about 3 Gyr shorter than the age determined for the Milky Way's globular clusters, which formed at an earlier epoch.

## Neutron Stars

### Neutron Degeneracy

### The Density of a Neutron Star

### The Equation of State

### Neutron Star Models

### The Chandrasekhar Limit for Neutron Stars

### Rapid Rotation and Conservation of Angular Momentum

### "Freezing In" Magnetic Field Lines

### Neutron Star Temperatures

## Pulsars

### General Characteristics

### Possible Pulsar Models

### Pulsars as Rapidly Rotating Neutron Stars

### Geminga

### Evidence for a Core-Collapse Supernova Origin

### Synchrotron and Curvature Radiation

### The Energy Source for the Crab's Synchrotron Radiation

### The Structure of the Pulses

### The Basic Pulsar Model

### Correlation Between Period Derivatives and Pulsar Classes

### Toward a Model of Pulsar Emission

### Magnetars and Soft Gamma Repeaters